tools/esdlcmd/README.md

```diff
@@ -0,0 +1,485 @@
+
+### Syntax
+
+These elements may create artifact content, change the tool's behavior, or both. When used as intended, none of these elements will appear in the generated output:
+
+#### Manifest
+Required root element of all manifest files that create output:
+
+- `bundle` output is created by default. It can be made explicit by setting either command line option `--output-type` to `bundle` and/or `@outputType` is `bundle`.
+- `binding` output is created when either command line option `--output-type` is `binding` and/or `@outputType` is `binding`.
+
+| Attribute | Required? | Value | Description |
+| - | :-: | - | - |
+| `@outputType` | N | string | A hint informing the tool which type of output to generate. The command line option `--output-type` may supersede this value to produce a different output.<br/>A `bundle` manifest is a superset of a `binding` manifest and may logically be used to create either output type. A `binding` manifest, as a subset of a `bundle`, cannot be used to create a valid `bundle`. |
+| `@xmlns[:prefix]` | Y | string | The manifest namespace `urn:hpcc:esdl:manifest` must be declared. The default namespace prefix should not be used unless all other markup is fully qualified. |
+
+#### ServiceBinding
+Recommended child of `Manifest` that creates ESDL binding content and applies ESDL binding-specific logic to descendent content:
+
+- A `Binding` output element is always created containing attributes of this element as described below. Attributes referenced by the tool are described in the table below.
+- A child of `Binding` named `Definition` will be created with attributes `@id` and `@esdlservice`.
+- Recognition of `<em:Scripts>` elements is enabled.
+- Recognition of `<em:Transform>` elements is enabled.
+
+There are three categories of attributes that can be defined on the `<em:ServiceBinding>` element. First are the standard attributes used to setup and define the binding, currently only `@esdlservice`:
+
+| Attribute   | Required? | Value  | Usage |
+| - | :-: | - | - |
+| `@esdlservice`  | Yes | string | - Name of the ESDL service to which the binding is bound. Output on the `Binding/Definition` element. <br/>- Used to generate a value for `Definition/@id` for `bundle` type output. Note that `Definition/@id` is not output by the tool for `binding` output as it would need to match the `ESDLDefinitionId` passed on command line `esdl bind-service` call, which may differ between environments. |
+
+Second are binding or service-specific attributes. This includes any attribute not in the other categories. These are all output on the `Binding/Definition` element. This category will likely grow over time but currently is just `@auth_feature`:
```

Comment:
Last sentence is incorrect. A check of `EsdlServiceImpl::configureTargets` reveals:
- namespace
- returnSchemaLocationOnOK

Reply:
added